### PR TITLE
Add site media attachments: photos, videos, and documents on dive sites

### LINKED ISSUE #211: Feature Request: Images for Dive sites
It would be great to support adding images directly to dive sites, not just individual dives.

Many dive sites have useful visual resources available online, such as maps, parking locations, and entry points. Being able to attach these to the dive site itself would make planning and navigation much easier.

### LINKED ISSUE #627: Dive sites / Dive site – Attachments or photos
I have drawn my own underwater maps for many of my dive sites and would like to upload them directly to the dive site so that I can access them later.

I can’t seem to find a way to add a photo or an attachment (e.g. a PDF) to the dive site.

### PR DESCRIPTION
**Pull request by ericgriffin**
## Summary

Dive sites can now hold their own media: entry-point photos, videos, hand-drawn underwater maps, and documents (PDFs render in-app; other formats open externally). Documents are also attachable to dives. The site detail page gains a Site Media section showing direct attachments plus a collapsed group of photos from dives logged at the site.

Closes #211
Closes #627

Design spec: `docs/superpowers/specs/2026-08-10-site-media-attachments-design.md`
Implementation plan: `docs/superpowers/plans/2026-08-10-site-media-attachments.md`

## Changes

- Reuse the existing `media.siteId` FK (already synced and merge-safe); add the missing read path: `getMediaForSite`, count and dedupe lookups, site media providers, and a `SiteMediaListNotifier`
- New `MediaType.document` with `isPdf`/`documentExtension` helpers and document MIME types in `shareMimeType` and `StoreKeys.contentTypeFor`
- Migration v148: `idx_media_site_id` query index plus site-side duplicate cleanup and partial unique index mirroring the dive-side v38 pair (self-guarded for partial migration-test fixtures)
- Correctness fix: site deletion now cascades media explicitly with HLC stamps (`partitionMediaForSiteDeletion` / `unlinkMediaFromDeletedSites`), mirroring the dive-side fix — the silent FK `SET NULL` wrote no HLC stamp and would have diverged peers once site media existed. Site-only media dies with the site through the deletion coordinator (tombstones + remote blob-delete intents); dive-linked and library-level rows survive unlinked
- Extracted shared grid pieces (`MediaSelectionHeader`, `MediaEmptyState`, `MediaThumbnailTile`, `OrphanedMediaPlaceholder`) out of `DiveMediaSection` into `media_grid.dart`; labels are constructor parameters so shared widgets stay l10n-free
- New `SiteMediaSection` (attachments grid with multi-select removal, add menu, collapsed dive-photos group) mounted on the site detail page
- New `SiteMediaViewerPage` (site-scoped fullscreen pager, no dive overlays) and `DocumentViewerPage` (in-app PDF viewer via pdfrx with share and unavailable/origin-device states)
- Site photo import flow reusing the photo picker without a dive time window; document attach flow links files by reference (security-scoped bookmark on iOS/macOS, persisted SAF URI on Android, plain path on Windows/Linux) — no byte copies; the media store upload is the durability path
- PDF thumbnails: page-1 render via pdfrx (512 px JPEG) wired into `ThumbnailGenerator` with an injectable renderer seam for hermetic tests; non-PDF documents show a generic tile with an extension badge
- Dive side: `DiveMediaSection` add button becomes a photos/document menu when wired, document tiles route to the document viewer, enrichment skips documents, and the dive photo viewer filters documents out of its pager
- New dependency: `pdfrx ^2.4.5` (pdfium-based; iOS/Android/macOS/Windows/Linux)
- All new user-facing strings localized in all 11 locales

## Test Plan

- [x] `flutter test` passes (16,104 tests, 17 skipped — includes 2 new PDF render tests gated on a host `PDFIUM_PATH`, verified locally against a real pdfium build)
- [x] `flutter analyze` passes (zero issues, whole project)
- [x] Manual testing on: macOS (photo/document attach, PDF viewing, site deletion cascade)

## Screenshots

Site Media section renders as a standard card between Marine Life and Difficulty on the site detail page; document tiles show a PDF/file icon with an extension badge.

### CODE REVIEW

File: lib/features/media/presentation/pages/site_media_viewer_page.dart [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+          }
+
+          // Find initial index
+          final initialIndex = mediaList.indexWhere(
+            (m) => m.id == widget.initialMediaId,
+          );
+          if (initialIndex != -1 && _pageController.hasClients == false) {
+            _currentIndex = initialIndex;
+            _pageController = PageController(initialPage: initialIndex);
+          }
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/media/presentation/pages/site_media_viewer_page.dart:109]: Fixed. `initState` created the controller and `build` overwrote it without disposing. Since the guard is `hasClients == false` no PageView is attached at that point, so it now disposes before reassigning.

Worth flagging that the same pattern is pre-existing in `trip_photo_viewer_page.dart` and `photo_viewer_page.dart` (this file was modelled on the trip one). I left those alone rather than widening this PR, but they have the same leak.

File: lib/features/media/presentation/widgets/media_grid.dart [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+                  child: const Icon(
+                    Icons.videocam,
+                    size: 16,
+                    color: Colors.white,
+                  ),
+                ),
+              ),
+
+            // Document extension badge (top-right, mirrors the video slot)
+            if (item.isDocument && !isSelected)
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/features/media/presentation/widgets/media_grid.dart]: Fixed — the badge is now gated on `item.documentExtension.isNotEmpty`, so an extensionless attachment renders no chip at all instead of an empty black one. Covered by a regression test in `media_grid_test.dart` ("an extensionless document draws no empty badge").

File: test/features/media/presentation/widgets/site_media_section_test.dart [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+          'site-1',
+        ).overrideWith((ref) async => const <MediaItem>[]),
+        mediaFromDivesAtSiteProvider(
+          'site-1',
+        ).overrideWith((ref) async => divePhotos),
+      ],
+      child: MaterialApp(
+        localizationsDelegates: AppLocalizations.localizationsDelegates,
+        supportedLocales: AppLocalizations.supportedLocales,
+        home: Scaffold(
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply test/features/media/presentation/widgets/site_media_section_test.dart]: Fixed — pinned `locale: const Locale('en')` on the `MaterialApp`. This is a real trap here, not theoretical: `flutter_test` forwards the HOST machine's locale list rather than a fixed en_US, and the app supports 11 locales, so `basicLocaleListResolution` hands a non-English contributor a translated UI and every English `find.text` returns nothing. CI stays green because its runners are en_US, so it would only bite locally. The whole file now goes through a shared harness that pins it.

File: test/features/media/presentation/widgets/dive_media_section_test.dart [hunk 4]
```diff
@@ -15,4 +22,62 @@ void main() {
...
+      overrides: [
+        ...base,
+        mediaForDiveProvider(
+          'dive-1',
+        ).overrideWith((ref) async => const <MediaItem>[]),
+      ],
+      child: MaterialApp(
+        localizationsDelegates: AppLocalizations.localizationsDelegates,
+        supportedLocales: AppLocalizations.supportedLocales,
+        home: Scaffold(
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply test/features/media/presentation/widgets/dive_media_section_test.dart]: Fixed — pinned `locale: const Locale('en')`. `flutter_test` forwards the HOST machine's locale list, so with 11 supported locales an unpinned MaterialApp renders translated UI on a non-English dev machine and every English `find.text` finds nothing (CI hides it: its runners are en_US). This file now builds through the shared media test harness, which pins the locale in one place.

File: test/features/media/presentation/pages/site_media_viewer_page_test.dart [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+      ProviderScope(
+        overrides: [
+          mediaForSiteProvider(
+            'site-1',
+          ).overrideWith((ref) async => const <MediaItem>[]),
+        ],
+        child: const MaterialApp(
+          localizationsDelegates: AppLocalizations.localizationsDelegates,
+          supportedLocales: AppLocalizations.supportedLocales,
+          home: SiteMediaViewerPage(
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply test/features/media/presentation/pages/site_media_viewer_page_test.dart:35]: Fixed on both occurrences — pinned `locale: Locale('en')`. `flutter_test` forwards the HOST machine's locale list, so with 11 supported locales an unpinned MaterialApp resolves to a translated UI on a non-English dev machine and the English assertions all miss; CI hides it because its runners are en_US. The expanded tests in this file now go through the shared harness, which pins it centrally.

File: test/features/media/presentation/pages/document_viewer_page_test.dart [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  overrides: [
+    resolvedFullResolutionProvider(item).overrideWith(
+      (ref) async =>
+          const ResolvedAssetResult(status: ResolutionStatus.unavailable),
+    ),
+  ],
+  child: MaterialApp(
+    localizationsDelegates: AppLocalizations.localizationsDelegates,
+    supportedLocales: AppLocalizations.supportedLocales,
+    home: DocumentViewerPage(item: item),
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply test/features/media/presentation/pages/document_viewer_page_test.dart:44]: Fixed — pinned `locale: const Locale('en')` on this file's `_host` helper, so every test in it (including the new loading/error/share cases) asserts against English regardless of the host machine's locale list.

**Comment by ericgriffin** [comment]
## Review feedback addressed + patch coverage raised to ~93%

All six inline comments are fixed and replied to individually. Each was verified against the codebase before changing anything — all three were real.

**Controller leak** — `initState` built the `PageController` and `build` overwrote it without disposing. Now disposes before reassigning (safe: the guard is `hasClients == false`). Note the same pattern is pre-existing in `trip_photo_viewer_page.dart` and `photo_viewer_page.dart`; left alone rather than widening this PR.

**Empty document badge** — `documentExtension` returns `''` for an extensionless filename, drawing an empty black chip. Gated on non-empty, with a regression test.

**Unpinned test locales** — real, and a documented trap here: `flutter_test` forwards the *host* machine's locale list, so with 11 supported locales an unpinned `MaterialApp` renders translated UI on a non-English machine and every English assertion misses. CI hides it (en_US runners).

### Coverage: 48.1% → **93.5%**

Measured locally by intersecting the branch diff with `lcov.info`, applying `codecov.yml`'s own `ignore:` rules so the number is comparable.

| File | Before | After |
| --- | --- | --- |
| `site_media_section.dart` | 43.8% | **100%** |
| `site_media_viewer_page.dart` | 17.8% | **97.1%** |
| `media_grid.dart` | 36.0% | **100%** |
| `media_item_view.dart` | 13.6% | **100%** |
| `site_media_providers.dart` | 77.8% | **100%** |
| `dive_media_section.dart` | 58.8% | **100%** |
| `media_repository.dart` | 93.5% | **100%** |
| `document_viewer_page.dart` | 53.8% | 75.4% |
| `pdf_page_renderer.dart` | 0% | 33.3% |

What unlocked it was a shared widget-test harness (`test/features/media/presentation/support/media_widget_harness.dart`): app base overrides + a fake `MediaSourceResolver` serving decodable bytes + a pinned locale. Media tiles could not render in widget tests before, which is why the grid paths were originally marked `coverage:ignore` — those markers are now gone and the paths are genuinely covered.

Also added: a v148 migration test (index creation, duplicate collapse, and the guard for fixtures whose `media` table predates `site_id`), and a locale test asserting every language actually carries the new strings rather than silently falling back to English.

One production change in service of testability: `SiteMediaImportHelper.linkSelectedAssets` splits the post-picker half (persist → invalidate → report) away from `showPhotoPicker`, which cannot run under `flutter_test`.

### What is still uncovered, and why

52 lines, most of them pdfium. `flutter_test` cannot load the pdfium binary, and a failed init crashes pdfrx's worker isolate so the calling future never completes — the test hangs to timeout rather than throwing, which a `try`/`catch` cannot rescue. Those paths stay behind the `PDFIUM_PATH` gate and were verified locally against a real pdfium build. The rest are platform-channel calls (`SharePlus`, `path_provider`) and the `FilePicker`-backed page closures.

`main` was merged into this branch remotely while I was working; I merged it down rather than force-pushing, regenerated codegen (main had added a mock), and re-ran the affected suites — 1,662 tests green, `flutter analyze` clean, `dart format` a no-op.

**Comment by ericgriffin** [comment]
On the two suppressed doc comments (`docs/FEATURE_ROADMAP.md:1230`, `docs/contributing/roadmap.md:239`) — pushing back on these rather than changing them, because the stated premise doesn't hold.

**Neither row introduces an emoji marker.** Both cells already had one; this PR only changes *which* one, because the feature shipped:

```
docs/FEATURE_ROADMAP.md:1230
-  | User-submitted site photos | 📋 Planned | v2.0 | Photo gallery per site |
+  | User-submitted site photos | ✅ Implemented | v1.5 | Site media section: ... |

docs/contributing/roadmap.md:239
-  - :crystal_ball: Site photo galleries
+  - :white_check_mark: Site photo galleries (site media section with ...)
```

**Both files are built entirely on emoji status markers** — 361 in `FEATURE_ROADMAP.md`, 136 shortcodes in `contributing/roadmap.md`, and the latter opens with a "Status Legend" table that *defines* `:white_check_mark:` / `:construction:` / `:clipboard:` / `:crystal_ball:` as its vocabulary. Dropping the icon from one bullet would break that legend contract for that row and leave two plain-text entries among ~500 iconed ones, which reads worse, not better.

The CLAUDE.md rule ("No emojis in code, comments, or documentation") is real, and these two files are wholesale at odds with it. Reconciling that is a docs-wide sweep touching ~500 lines and both files' legends — worth doing deliberately, but not as a drive-by in a feature PR about site media. Happy to open a separate issue for it if that's wanted.

Everything else from the review round is fixed and pushed (see the previous comment).

File: lib/core/database/database.dart [hunk 7]
```diff
@@ -7664,6 +7668,45 @@ class AppDatabase extends _$AppDatabase {
...
+                SELECT platform_asset_id, site_id, MIN(created_at) as min_created
+                FROM media
+                WHERE platform_asset_id IS NOT NULL AND site_id IS NOT NULL
+                GROUP BY platform_asset_id, site_id
+                HAVING COUNT(*) > 1
+              ) dupes ON m.platform_asset_id = dupes.platform_asset_id
+                AND m.site_id = dupes.site_id
+                AND m.created_at > dupes.min_created
+            )
+          ''');
```
[the remark this answers is not in the document]
Comment (ericgriffin) [reply lib/core/database/database.dart:7705]: Good catch — this was a real bug, and worse than flaky: the `CREATE UNIQUE INDEX` immediately after would throw `UNIQUE constraint failed`, which aborts `onUpgrade` and leaves the database unopenable.

Reproduced it first (two rows tied at the oldest `created_at`):

```
SqliteException(2067): UNIQUE constraint failed: media.platform_asset_id, media.site_id
  Causing statement: CREATE UNIQUE INDEX IF NOT EXISTS idx_media_asset_site_unique
```

The tie is reachable in practice, not just in theory: `created_at` is epoch **milliseconds** and a bulk import writes many rows inside one, and a site merge repoints rows carrying the same `platform_asset_id` onto the survivor site — which is exactly how two rows end up sharing a (platform_asset_id, site_id) pair in the first place.

Fixed with the `ROW_NUMBER()` pattern you pointed at (the `deletion_log` cleanup at line ~3563), ordered `created_at ASC, rowid ASC` so exactly one row survives per pair. One wrinkle worth noting: the outer `DELETE` has to stay scoped to rows where both columns are non-null, because the subquery deliberately excludes unlinked rows and a bare `rowid NOT IN (...)` would otherwise sweep every one of them.

Regression test added in `migration_v148_site_media_test.dart` — asserts a single survivor *and* that the unique index exists, so a future tie-blind rewrite fails loudly instead of silently aborting the upgrade.

One thing to flag: the dive-side v38 cleanup this was modelled on has the same tie-blindness. I left it alone since it only affects databases still below v38, but it is the same latent failure if you want it swept separately.
